src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,238 @@
+                    SyntaxFactory.Token(
+                        binaryExpression.OperatorToken.LeadingTrivia,
+                        negatedOperatorType,
+                        binaryExpression.OperatorToken.TrailingTrivia),
+                    binaryExpression.Right);
+
+                return true;
+            }
+
+            result = null;
+            return false;
+        }
+
+        private ExpressionSyntax Negate(ExpressionSyntax expression, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (TryNegateBinaryComparisonExpression(expression, semanticModel, cancellationToken, out var result))
+            {
+                return result;
+            }
+
+            switch (expression.Kind())
+            {
+                case SyntaxKind.ParenthesizedExpression:
+                {
+                    var parenthesizedExpression = (ParenthesizedExpressionSyntax)expression;
+                    return parenthesizedExpression
+                        .WithExpression(Negate(parenthesizedExpression.Expression, semanticModel, cancellationToken))
+                        .WithAdditionalAnnotations(Simplifier.Annotation);
+                }
+
+                case SyntaxKind.LogicalNotExpression:
+                {
+                    var logicalNotExpression = (PrefixUnaryExpressionSyntax)expression;
+
+                    var notToken = logicalNotExpression.OperatorToken;
+                    var nextToken = logicalNotExpression.Operand.GetFirstToken(
+                        includeZeroWidth: true, includeSkipped: true, includeDirectives: true, includeDocumentationComments: true);
+
+                    var existingTrivia = SyntaxFactory.TriviaList(
+                        notToken.LeadingTrivia.Concat(
+                            notToken.TrailingTrivia.Concat(
+                                nextToken.LeadingTrivia)));
+
+                    var updatedNextToken = nextToken.WithLeadingTrivia(existingTrivia);
+
+                    return logicalNotExpression.Operand.ReplaceToken(
+                        nextToken,
+                        updatedNextToken);
+                }
+
+                case SyntaxKind.LogicalOrExpression:
+                {
+                    var binaryExpression = (BinaryExpressionSyntax)expression;
+                    result = SyntaxFactory.BinaryExpression(
+                        SyntaxKind.LogicalAndExpression,
+                        Negate(binaryExpression.Left, semanticModel, cancellationToken),
+                        SyntaxFactory.Token(
+                            binaryExpression.OperatorToken.LeadingTrivia,
+                            SyntaxKind.AmpersandAmpersandToken,
+                            binaryExpression.OperatorToken.TrailingTrivia),
+                        Negate(binaryExpression.Right, semanticModel, cancellationToken));
+
+                    return result
+                        .Parenthesize()
```

Comment:
do we have tests involving precedence changes and thus these parens shoudl/shouldn't be removed accordingly?